Common/src/main/java/com/techarium/techarium/inventory/MachineCoreMenu.java

```diff
@@ -0,0 +1,55 @@
+package com.techarium.techarium.inventory;
+
+import com.techarium.techarium.blockentity.multiblock.MachineCoreBlockEntity;
+import com.techarium.techarium.platform.CommonServices;
+import com.techarium.techarium.registry.TechariumMenuTypes;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+
+import java.util.List;
+
+public class MachineCoreMenu extends AbstractContainerMenu {
+
+	private final BlockPos pos;
+	private final MachineCoreBlockEntity machineCore;
+
+	public MachineCoreMenu(int id, Inventory playerInventory, Player player, BlockPos pos) {
+		super(TechariumMenuTypes.MACHINE_CORE.get(), id);
+		this.pos = pos;
+		this.machineCore = (MachineCoreBlockEntity) player.level.getBlockEntity(this.pos);
+	}
+
+	@Override
+	public ItemStack quickMoveStack(Player player, int index) {
+		return ItemStack.EMPTY;
+	}
+
+	@Override
+	public boolean stillValid(Player player) {
+		return player.distanceToSqr(this.pos.getX(), this.pos.getY(), this.pos.getZ()) <= 16.0D;
```

Comment:
Should be 64, to make the distance 8. Because otherwise the isValid distance would be lower than the default reach range.